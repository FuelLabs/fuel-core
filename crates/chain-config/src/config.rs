mod blob;
mod chain;
mod coin;
mod consensus;
mod contract;
mod message;
#[cfg(feature = "test-helpers")]
mod randomize;
#[cfg(feature = "std")]
mod snapshot_metadata;
mod state;
mod table_entry;

pub use blob::*;
pub use chain::*;
pub use coin::*;
pub use consensus::*;
pub use contract::*;
pub use message::*;
#[cfg(feature = "test-helpers")]
pub use randomize::*;
#[cfg(feature = "std")]
pub use snapshot_metadata::*;
pub use state::*;
pub use table_entry::*;
