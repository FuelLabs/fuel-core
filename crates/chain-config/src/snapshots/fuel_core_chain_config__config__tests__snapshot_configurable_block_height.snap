---
source: crates/chain-config/src/config.rs
expression: json
---
{
  "chain_name": "local_testnet",
  "block_gas_limit": 1000000000,
  "initial_state": {
    "height": "0x0000000014c8be1f"
  },
  "transaction_parameters": {
    "contract_max_size": 16777216,
    "max_inputs": 255,
    "max_outputs": 255,
    "max_witnesses": 255,
    "max_gas_per_tx": 100000000,
    "max_script_length": 1048576,
    "max_script_data_length": 1048576,
    "max_storage_slots": 255,
    "max_predicate_length": 1048576,
    "max_predicate_data_length": 1048576,
    "gas_price_factor": 1000000000,
    "gas_per_byte": 4,
    "max_message_data_length": 1048576
  },
  "gas_costs": {
    "add": 1,
    "addi": 1,
    "aloc": 1,
    "and": 1,
    "andi": 1,
    "bal": 155,
    "bhei": 1,
    "bhsh": 1,
    "burn": 35,
    "cb": 2,
    "cfei": 1,
    "cfsi": 1,
    "croo": 163,
    "div": 1,
    "divi": 1,
    "ecr": 1601,
    "eq": 1,
    "exp": 1,
    "expi": 1,
    "flag": 1,
    "gm": 1,
    "gt": 1,
    "gtf": 1,
    "ji": 1,
    "jmp": 1,
    "jne": 1,
    "jnei": 1,
    "jnzi": 1,
    "k256": 18,
    "lb": 1,
    "log": 40,
    "lt": 1,
    "lw": 1,
    "mcpi": 3,
    "mint": 34,
    "mlog": 1,
    "mod": 1,
    "modi": 1,
    "move": 1,
    "movi": 1,
    "mroo": 2,
    "mul": 1,
    "muli": 1,
    "noop": 1,
    "not": 1,
    "or": 1,
    "ori": 1,
    "ret_contract": 61,
    "rvrt_contract": 62,
    "s256": 5,
    "sb": 1,
    "scwq": 10,
    "sll": 1,
    "slli": 1,
    "srl": 1,
    "srli": 1,
    "srw": 160,
    "sub": 1,
    "subi": 1,
    "sw": 1,
    "sww": 80,
    "swwq": 69,
    "time": 1,
    "tr": 113,
    "tro": 94,
    "xor": 1,
    "xori": 1,
    "call": {
      "base": 311,
      "dep_per_unit": 14
    },
    "ccp": {
      "base": 155,
      "dep_per_unit": 15
    },
    "csiz": {
      "base": 148,
      "dep_per_unit": 16
    },
    "ldc": {
      "base": 156,
      "dep_per_unit": 14
    },
    "logd": {
      "base": 47,
      "dep_per_unit": 21
    },
    "mcl": {
      "base": 1,
      "dep_per_unit": 2539
    },
    "mcli": {
      "base": 1,
      "dep_per_unit": 2610
    },
    "mcp": {
      "base": 1,
      "dep_per_unit": 1377
    },
    "meq": {
      "base": 1,
      "dep_per_unit": 2802
    },
    "retd_contract": {
      "base": 64,
      "dep_per_unit": 21
    },
    "smo": {
      "base": 79,
      "dep_per_unit": 14
    },
    "srwq": {
      "base": 424,
      "dep_per_unit": 6
    }
  },
  "consensus": {
    "PoA": {
      "signing_key": "22ec92c3105c942a6640bdc4e4907286ec4728e8cfc0d8ac59aad4d8e1ccaefb"
    }
  }
}
